oci/table_oci_core_vnic_attachment.go

```diff
@@ -0,0 +1,423 @@
+package oci
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/oracle/oci-go-sdk/v44/common"
+	"github.com/oracle/oci-go-sdk/v44/core"
+	"github.com/turbot/go-kit/helpers"
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableCoreVnicAttachment(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_core_vnic_attachment",
+		Description: "OCI Core VNIC Attachment",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("id"),
+			Hydrate:    getVnicAttachment,
+		},
+		List: &plugin.ListConfig{
+			Hydrate:           listVnicAttachments,
+			ShouldIgnoreError: isNotFoundError([]string{"404"}),
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "availability_domain",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "compartment_id",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "instance_id",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItem: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "id",
+				Description: "The OCID of the VNIC attachment.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "display_name",
+				Description: "A user-friendly name for the VNIC attachment.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "instance_id",
+				Description: "The OCID of the instance.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "availability_domain",
+				Description: "The availability domain of the instance.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "lifecycle_state`",
+				Description: "The current state of the VNIC attachment. Possible values include: 'ATTACHING', 'ATTACHED', 'DETACHING', 'DETACHED'.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "private_ip",
+				Description: "The private IP address of the primary `privateIp` object on the VNIC.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getVnic,
+				Transform:   transform.FromField("PrivateIp"),
+			},
+			{
+				Name:        "public_ip",
+				Description: "The public IP address of the VNIC, if one is assigned.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getVnic,
+				Transform:   transform.FromField("PublicIp"),
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the VNIC attachment was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
```

Comment:
Can we use the We do not need any transformation?

Reply:
`TimeCreated` property returns struct `common.SDKTime` which has property `time.Time`. Without transformation we will have the following error:
`Error: interfaceToColumnValue failed for column 'time_created': cannot convert 2022-02-08 10:01:36.871 +0000 UTC to a time.Time`